centreon/src/Core/HostTemplate/Application/UseCase/PartialUpdateHostTemplate/PartialUpdateHostTemplate.php

```diff
@@ -297,25 +238,15 @@ private function findOriginalMacros(HostTemplate $hostTemplate): array
     {
         $templateParents = $this->readHostTemplateRepository->findParents($hostTemplate->getId());
         $hostInheritance = new HostInheritance();
-        $inheritanceLineIds = $hostInheritance::findInheritanceLine($hostTemplate->getId(), $templateParents);
-
-        $existingHostMacros = $this->readHostMacroRepository->findByHostIds($inheritanceLineIds);
-        /** @var array<string,HostMacro> */
-        $inheritedMacros = [];
-        /** @var array<string,HostMacro> */
-        $directMacros = [];
-        foreach ($inheritanceLineIds as $id) {
-            foreach ($existingHostMacros as $macro) {
-                if ($macro->getHostId() === $hostTemplate->getId()) {
-                    $directMacros[$macro->getName()] = $macro;
-                } else if (
-                    ! isset($inheritedMacros[$macro->getName()])
-                    && $macro->getHostId() === $id
-                ) {
-                    $inheritedMacros[$macro->getName()] = $macro;
-                }
-            }
-        }
+        $inheritanceLine = $hostInheritance::findInheritanceLine($hostTemplate->getId(), $templateParents);
```

Comment:
Static call + **remove previous line** about `$hostInheritance`
```suggestion
        $inheritanceLine = HostInheritance::findInheritanceLine($hostTemplate->getId(), $templateParents);
```